### ISSUE TITLE
Changing tic-base is unsafe

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#741]
Changing the tic-base of the NEST `Time` class, e.g.,
```
0 << /tics_per_ms 1024. /resolution 0.5 >> SetStatus
```
invalidates the recording interval of voltmeter and multimeter, e.g.,
```
/voltmeter GetDefaults /interval get =
0.976562
```
The reason for this is that the interval is represented internally as a `Time` object, i.e., in tics; when the size of the tic changes, the value of the interval changes as well.

Any other `Time` objects stored in neuron and synapse prototypes will be affected as well, as may be `Time` objects held by managers etc.

Furthermore, `/tics_per_ms` requires a double argument, even though this clearly needs to be an integer value.

**Comment by heplesser** [comment]
As a side effect of the implicit change in the `interval`, the default-resetting implied in the following `Connect` with parameters leads to a confusing error:
```
SLI ] /voltmeter << /interval 1. >> Create 
May 31 23:11:55 SetDefaults_l_D [Error]: BadProperty
    Setting status of model 'voltmeter': The sampling interval must be a multiple of the simulation resolution
```

**Comment by heplesser** [comment]
## Proposal for solution

1. We can without practical problems restrict changes in tic-base to occur in a "fresh" kernel only, i.e., either a newly started kernel or immediately after `ResetKernel`.
1. In particular, we prohibit change of tic base after any `CopyModel`, `SetDefaults`, `Create`, `Connect` and `Simulate` calls
1. Thus, we only allow changes to tic base in situations where
    a. time is zero
    a. no neurons or synapses exist
    a. all node and synapse prototypes are identical to the corresponding pristine prototypes, and there is exactly one prototype per pristine prototype.
1. Therefore, we can only need to recalibrate any `Time` objects in
    a. kernel managers
    a. pristine prototypes
   and then recreate the prototypes from pristine prototypes
1. All model classes thus need to provide a method that recalibrates their `Time` objects
1. Essentially, this is an extension of the time-recalibration in `SimulationManager::set_status()` based on `TimeConverter`.

I suggest we address this once #480 has been merged.

**Comment by heplesser** [comment]
@hakonsbm It looks like your commit f681ca31adcb91 may cause problems, see this failing build https://travis-ci.org/nest/nest-simulator/jobs/636604220?utm_medium=notification&utm_source=github_status

**Comment by hakonsbm** [comment]
@heplesser Fixed by d528f86ca21dcb4f3e69ac66454b73ce37e15dc3.
